modules/cnf-about-topology-aware-lifecycle-operator-config.adoc

```diff
@@ -0,0 +1,127 @@
+* Pre- and post-update actions
+* Update timing
+
+The `ClusterGroupUpgrade` CR can be in the following states:
+
+`UpgradeNotStarted`:: This is the initial state of the `ClusterGroupUpgrade` CR.
++
+The Operator builds a remediation plan based on the following fields:
++
+* The `clusterSelector` field specifies the labels of the clusters that you want to update
+* The `clusters` field specifies a list of clusters to update
+* The `canaries` fields specifies the clusters for canary updates 
+* The `maxConcurrency` field specifies the number of clusters to update in a batch
++
+The `ClusterGroupUpgrade` CR transitions to the `UpgradeNotCompleted` state after the remediation plan is successfully completed and after the `enable` field is set to `true`. At this point, the Operator starts to update the `NonCompliant` clusters with the specified managed policies.
++
+[NOTE]
+====
+You can only make changes to the `spec` fields if the `ClusterGroupUpgrade` CR is either in the `UpgradeNotStarted` or the `UpgradeCannotStart` state. 
+====
++
+.Sample CR
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu
+  namespace: default
+spec:
+  clusters: <1>
+  - spoke1
+  - spoke2
+  - spoke3
+  enable: false
+  managedPolicies:
+  - policy1-common-cluster-version-policy
+  - policy2-common-pao-sub-policy
+  - policy3-common-ptp-sub-policy
+  remediationStrategy: <2>
+    canaries: <3>
+    - spoke1
+    maxConcurrency: 2 <4>
+    timeout: 240
+status: <5>
+  conditions:
+  - message: The ClusterGroupUpgrade CR is not enabled
+    reason: UpgradeNotStarted
+    status: "False"
+    type: Ready
+----
+<1> The list of clusters to update.
+<2> The `remediationStrategy` fields define the specifics of the cluster updates.
+<3> The remediation plan starts with the clusters listed in the `canaries` field. Each canary cluster forms a single-cluster batch.
+<4> The `maxConcurrency` field defines the maximum number of concurrent updates in a batch. The number of remediation batches is the number of canary clusters, plus the number of clusters (except the canary clusters) divided by `maxConcurrency`. The clusters that are already `Compliant`` with all the managed policies are excluded from the remediation plan.
+<5> The `status` field displays information about the status of the updates.
+
+`UpgradeNotCompleted`:: In this state, the Operator enforces the policies following the remediation plan defined in the `UpgradeNotStarted` state.
++
+Enforcing the policies of subsequent batches starts immediately after all the clusters of the current batch are `Compliant` with all the managed policies. If the batch times out, the Operator moves on to the next batch. The timeout value of a batch  is the `spec.timeout` field divided by the number of batches in the remediation plan.
++
+[NOTE]
+====
+The managed policies are applied in the order they are listed in the `managedPolicies` field in the `ClusterGroupUpgrade` CR.
```

Comment:
What do you think?
> The managed policies are applied in the order they are listed in the `managedPolicies` field in the `ClusterGroupUpgrade` CR. One managed policy is applied at a time. After the specified clusters become compliant to the current policy, the next managed policy on the list will be applied.

@bobfuru I think it would make sense to use future tense here. What's your take on it?

Reply:
> The managed policies are applied in the order they are listed in the managedPolicies field in the ClusterGroupUpgrade CR. One managed policy is applied at a time. After the specified clusters become compliant to the current policy, the next managed policy for which the cluster is NocCompliant will be applied.